packages/hydrooj/src/handler/problem.ts

```diff
@@ -793,6 +794,32 @@ export class ProblemFilesHandler extends ProblemDetailHandler {
         });
         this.response.redirect = this.url('record_detail', { rid });
     }
+
+    async postSyncData() {
+        this.checkPriv(PRIV.PRIV_EDIT_SYSTEM);
+
+        // push the sync data task to judgers
+        (async () => {
+            const doneIds = [];
+            let taskId: string | null = null;
+            do {
+                // eslint-disable-next-line no-await-in-loop
+                taskId = await this.ctx.serial('problem/syncData', this.pdoc.domainId, this.pdoc.docId, this.pdoc.data, doneIds);
```

Comment:
`serial` only iterates listeners on local process, and won't talk to other web processes, `emit` also the same.
only `broadcast` will send the event to everyone.

Reply:
Are there any ways to get all the websocket connections? It seems that using the database is necessary

Reply:
Is using a new type of `document` acceptable? Or reused the infomation in `status` collection?

Reply:
all web servers are intended to be stateless for horizontal scaling;
I'm thinking maybe adding a `autoSync: string[]` (`host/domainId/pid`) to judge config file might be a better solution and easily configurable using deployment tools like nix-deploy, ansible, or even system image.
Using server-push means those judges who joined later (scaled up) might get the sync event.